### ISSUE TITLE
Code Samples

### ISSUE THREAD

**Issue by ranielzed** [square/svelte-store#10]
Thanks for this great tool! It's a big help in organizing things and makes it easier to manage and reason about.

Just wondering what's the recommended approach for handling individual items in a store?
For example, updating a single item stored inside an AsyncWritable store, sending it back to the server, etc. instead of a whole list of items.
I've noticed the example of removing an item from a store and it would be nice if there's also an example of how to use it when adding / updating items.

Thanks

**Comment by Akolyte01** [comment]
Writing is a complex topic because how it needs to be preformed is going to change a lot depending on the design of the endpoints involved. Different endpoints may have very different ways to represent updating data. As a result this package isn't capable of supporting every relationship. So when writing corresponding stores you may need to write more boilerplate, or even write some custom stores. 

For example, lets consider an endpoint setup where we can read a collection of items mapped by an id e.g.
```
{
  a: { id: 'a', content: 'foo' },
  b: { id: 'b', content: 'bar'},
}
```
Our endpoint for updating this however takes just one of the items, with updated content. 
In such a circumstance the built in asynchronous writing features of this package aren't very useful for us. However we can always add our own custom functionality.

```
// we make an asyncWritable store, but don't include a mappingWriteFunction
// this will mean we can update the store's value, just like a regular writable store
// however it won't kick off any async behavior on its own. 
const myItemMap = asyncWritable(
  [],
  async () => fetchItemMap(),
);
 
// then we handle updating ourselves
const updateItemContent = (id, newContent) => {
   myItemMap.update((itemMap) => {
     itemMap[id] = { id, content: newContent };
     return itemMap;
   }); 
  persistContentChangeToBackend(id, newContent);
}
```

Once we have this logic built out we can go one step further and fold this into a custom store.

```
const _myItemMap = asyncWritable(
  [],
  async () => fetchItemMap(),
);

export const myItemMap = { 
  subscribe: _myItemMap.subscribe,
  load: _myItemMap.load,
  update: (id, newContent) => {
    _myItemMap.update((itemMap) => {
      itemMap[id] = { id, content: newContent };
      return itemMap;
    }); 
    persistContentChangeToBackend(id, newContent);
  }
}
```

So here we have an internal store `_myItemMap` with the default set of properties of an asyncWritable store. Then we export a different version of the store with only the properties that we want consumers to use. In this case, we include an 'update' function to replace the default update function that will handle both updating the value of the store and persisting changes to the backend. So now you can call `myItemMap.update('a', 'some new content')`



So, TL;DR, you may need to handle some boilerplate yourself, unfortunately. Custom stores can help, particularly if you need to perform the same kind of action for multiple endpoints. More info on custom stores can be found in a talk I gave at the spring svelte summit, if you haven't seen it  https://youtu.be/qqj2cBockqE?t=3980

**Comment by sanderjson** [comment]
I have one question about the above example.

Let's say I have multiple stores following the pattern above each with their own unique route for fetch. But I also have a single dedicated route to fetch all data for initial load.

Can I use that single get all route to set all the other stores _without_ triggering their individual fetch requests?

Essentially I want to leave each async store syncable to a unique url but be able to initially seed them from a single request.

**Comment by Akolyte01** [comment]
What's the intent behind fetching the data in different ways at
different times in your apps lifecycle? Offhand I would recommend either
using the single endpoint to hydrate your application and using derived
stores to split from there into the different pieces of data you need, or
to use the individual routes from the get go.
